### PR TITLE
Eyebot police activity

### PR DESCRIPTION
**Pull request by nexusmrsep**
#### Summary
```SUMMARY: Features "Eyebots speak to suspects, doing police activity"```

#### Purpose of change
@DracoGriffin recently said on Reddit that eyebot police talk would be nice, and I thought, "yeah, that's about spot on"

#### Describe the solution

Eyebots audibly interact with suspects when taking their picture:
- since eyebot takes pictures, it recognizes you in police database and calls you by name (can be offline, can be incorrect)
- if you hold a gun you are called to drop it
- if you hold a weapon are called to drop it
- if you are cuffed you are ignored, go to jail sucker
- more random police talk follows

Speech lines are JSON-ized, can be easily expanded later.

#### Describe alternatives you've considered
There is certainly potential here for more interactions, like danger assessment and equivalent force reaction etc., but that is beyond the scope of this PR

#### Additional context
Police brutality!
image

### CODE REVIEW

File: src/monattack.cpp [hunk 1]
```diff
@@ -2785,14 +2786,32 @@ bool mattack::photograph( monster *z )
...
+        // Friendly (hacked?) bot ignore the player. Arrested suspect ignored too.
         // TODO: might need to be revisited when it can target npcs.
         return false;
     }
     z->moves -= 150;
     add_msg( m_warning, _( "The %s takes your picture!" ), z->name().c_str() );
     // TODO: Make the player known to the faction
+    if( one_in( 6 ) ) {
+        sounds::sound( z->pos(), 15, sounds::sound_t::alert,
+                       _( "a robotic voice boom, \"Citizen " ) + Name::generate( g->u.male ) + "!\"" );
```
Comment (AMurkin) [inline src/monattack.cpp]: `string_format` would be better for I18N:
```cpp
string_format( _( "a robotic voice boom, \"Citizen  %s!\"" ), Name::generate( g->u.male ) )
```

File: data/json/speech.json [hunk 2]
```diff
@@ -1259,5 +1259,140 @@
...
+        "volume" : 15
+    },{
+        "type" : "speech",
+        "speaker" : "mon_eyebot",
+        "sound" : "\"Remain where you are!\"",
+        "volume" : 15
+    },{
+        "type" : "speech",
+        "speaker" : "mon_eyebot",
+        "sound" : "\"Police inbound. Stay where you are!\"",
```
Comment (AMurkin) [inline data/json/speech.json]:
```suggestion
        "sound" : "\"Police inbound.  Stay where you are!\"",
```

File: src/monattack.cpp (hunk 1 above)
Comment (thquinn) [inline src/monattack.cpp]: a robotic voice *booms*

Comment (nexusmrsep) [reply src/monattack.cpp]: Are you sure? Player side would be "You hear a robotic voice booms, XXX" instead of "You hear a robotic voice boom, XXX"

Comment (thquinn) [reply src/monattack.cpp]: Yeah, never mind. I should familiarize myself more before I open my mouth. :)

Comment (AMurkin) [reply src/monattack.cpp]: That's for strings with variables only. No need to do it for plain text strings.

Comment (AMurkin) [reply src/monattack.cpp]: Please take a closer look at my example. `string_format` have several arguments: format string and the values for the placeholders.

Comment (nexusmrsep) [reply src/monattack.cpp]: should have paid more attention to it from the beginning, thanks

File: src/monattack.cpp [hunk 3]
```diff
@@ -2785,14 +2786,32 @@ bool mattack::photograph( monster *z )
...
+        // Friendly (hacked?) bot ignore the player. Arrested suspect ignored too.
         // TODO: might need to be revisited when it can target npcs.
         return false;
     }
     z->moves -= 150;
     add_msg( m_warning, _( "The %s takes your picture!" ), z->name().c_str() );
     // TODO: Make the player known to the faction
+    if( one_in( 6 ) ) {
+        sounds::sound( z->pos(), 15, sounds::sound_t::alert,
+                       string_format( _( "a robotic voice boom, \"Citizen %s!" ) ) );
```
Comment (AMurkin) [inline src/monattack.cpp]:
```suggestion
                       string_format( _( "a robotic voice boom, \"Citizen %s!\"",
                                      Name::generate( g->u.male ) ) ) );
```

File: src/monattack.cpp [hunk 4]
```diff
@@ -2785,14 +2786,32 @@ bool mattack::photograph( monster *z )
...
+    } else if( one_in( 3 ) ) {
+        sounds::sound( z->pos(), 15, sounds::sound_t::alert,
+                       string_format( _( "a robotic voice boom, \"Citizen %s!" ) ) );
```
Comment (AMurkin) [inline src/monattack.cpp]:
```suggestion
                       string_format( _( "a robotic voice boom, \"Citizen %s!\"",
                                      g->u.name ) ) );
```

File: src/monattack.cpp [hunk 5]
```diff
@@ -2785,14 +2786,32 @@ bool mattack::photograph( monster *z )
...
-        // Friendly (hacked?) bot ignore the player.
+    if( z->friendly || g->u.weapon.typeId() == "e_handcuffs" ) {
+        // Friendly (hacked?) bot ignore the player. Arrested suspect ignored too.
         // TODO: might need to be revisited when it can target npcs.
         return false;
     }
     z->moves -= 150;
     add_msg( m_warning, _( "The %s takes your picture!" ), z->name().c_str() );
     // TODO: Make the player known to the faction
+    if( one_in( 6 ) ) {
```
Comment (ZhilkinSerg) [inline src/monattack.cpp]: I would reduce this code block to something like this:

```CPP
    const std::string name = "...  database connection lost";
    if( one_in( 6 ) ) {
        name = Name::generate( g->u.male );
    } else if( one_in( 3 ) ) {
        name = g->u.name;
    }
    sounds::sound( z->pos(), 15, sounds::sound_t::alert, string_format( _( "a robotic voice boom, \"Citizen %s!\"" ), name ) );
```

Comment (nexusmrsep) [reply src/monattack.cpp]: Done.

**Comment by adamkad1** [comment]
what responses it will have to 'authorized' people? like policemen? maybe like 'Officer [insert name here]!' 
Keep up the good work!

**Comment by nexusmrsep** [comment]
@adamkad1 Those were already in. Something along the lines of "human officer on scene", "SWAT team deployed" etc.

**Pull request merged**